src/cloud-api-adaptor/azure/cmd/copy-image/main.go

```diff
@@ -0,0 +1,188 @@
+
+	galleriesClient, err := armcompute.NewGalleriesClient(subscriptionID, cred, nil)
+	if err != nil {
+		log.Fatalf("creating galleries client: %v", err)
+	}
+	galPoller, err := galleriesClient.BeginCreateOrUpdate(ctx, resourceGroup, galleryName, armcompute.Gallery{Location: to.Ptr(location)}, nil)
+	if err != nil {
+		log.Fatalf("creating gallery: %v", err)
+	}
+	if _, err = galPoller.PollUntilDone(ctx, nil); err != nil {
+		log.Fatalf("waiting for gallery: %v", err)
+	}
+
+	imgDefClient, err := armcompute.NewGalleryImagesClient(subscriptionID, cred, nil)
+	if err != nil {
+		log.Fatalf("creating gallery images client: %v", err)
+	}
+	imgDefPoller, err := imgDefClient.BeginCreateOrUpdate(ctx, resourceGroup, galleryName, imageDefinitionName, armcompute.GalleryImage{
+		Location: to.Ptr(location),
+		Properties: &armcompute.GalleryImageProperties{
+			OSType:           to.Ptr(armcompute.OperatingSystemTypesLinux),
+			OSState:          to.Ptr(armcompute.OperatingSystemStateTypesGeneralized),
+			HyperVGeneration: to.Ptr(armcompute.HyperVGenerationV2),
+			Identifier: &armcompute.GalleryImageIdentifier{
+				Publisher: to.Ptr("cvm-publisher"),
+				Offer:     to.Ptr("cvm-offer"),
+				SKU:       to.Ptr("cvm-sku"),
+			},
+			Features: []*armcompute.GalleryImageFeature{{
+				Name:  to.Ptr("SecurityType"),
+				Value: to.Ptr("ConfidentialVmSupported"),
+			}},
+		},
+	}, nil)
+	if err != nil {
+		log.Fatalf("creating image definition: %v", err)
+	}
+	if _, err = imgDefPoller.PollUntilDone(ctx, nil); err != nil {
+		log.Fatalf("waiting for image definition: %v", err)
+	}
+
+	versionClient, err := armcompute.NewGalleryImageVersionsClient(subscriptionID, cred, nil)
+	if err != nil {
+		log.Fatalf("creating gallery image versions client: %v", err)
+	}
+	verPoller, err := versionClient.BeginCreateOrUpdate(ctx, resourceGroup, galleryName, imageDefinitionName, imageVersionName, armcompute.GalleryImageVersion{
+		Location: to.Ptr(location),
+		Properties: &armcompute.GalleryImageVersionProperties{
+			PublishingProfile: &armcompute.GalleryImageVersionPublishingProfile{TargetRegions: targets},
+			StorageProfile: &armcompute.GalleryImageVersionStorageProfile{
+				Source: &armcompute.GalleryArtifactVersionFullSource{ID: to.Ptr(imageID)},
+			},
+		},
+	}, nil)
+	if err != nil {
+		log.Fatalf("creating image version: %v", err)
+	}
+	if _, err = verPoller.PollUntilDone(ctx, nil); err != nil {
+		log.Fatalf("waiting for image version: %v", err)
+	}
+
+	log.Printf("deleting temporary managed image %s", userImageName)
+	delImgPoller, err := imagesClient.BeginDelete(ctx, resourceGroup, userImageName, nil)
+	if err == nil {
+		_, _ = delImgPoller.PollUntilDone(ctx, nil)
+	}
+
+	log.Printf("deleting temporary managed disk %s", managedDiskName)
+	delDiskPoller, err := diskClient.BeginDelete(ctx, resourceGroup, managedDiskName, nil)
+	if err == nil {
+		_, _ = delDiskPoller.PollUntilDone(ctx, nil)
+	}
```

Comment:
please use defer() to delete those always, also in case of errors when creating an image version in the user's gallery